### ISSUE TITLE
Alternative Loss Functions

### ISSUE THREAD

**Issue by cfoster0** [cfoster0/CLAP#16]
An alternative to the symmetric cross entropy loss, we could optimize for alignment and uniformity on the unit hypersphere as was done in this paper:

https://arxiv.org/abs/2005.10242

The code itself is like 4 lines of PyTorch, with the caveat that we'd need to write a `pdist` JAX function in order to use it.

**Comment by cfoster0** [comment]
May be worthwhile to play with an additional speech-domain loss in the form of DINO, which should be relatively easy to implement, with a bit of JAX code and some crop augmentations.
